src/textinput/TextInput.ts

```diff
@@ -0,0 +1,37 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { FormLabelMixin, FormLabelMixinProperties } from '@dojo/widget-core/mixins/FormLabel';
+import { v } from '@dojo/widget-core/d';
+import * as css from './styles/textinput.css';
+
+/**
+ * @type TextInputProperties
+ *
+ * Properties that can be set on a TextInput component
+ *
+ * @property type			Input type, e.g. text, email, tel, etc.
+ * @property onInput	Called when the node's 'input' event is fired
+ */
+export interface TextInputProperties extends ThemeableProperties, FormLabelMixinProperties {
+	type?: string;
+	onInput?(event: Event): void;
+}
+
+@theme(css)
+export default class TextInput extends ThemeableMixin(FormLabelMixin(WidgetBase))<TextInputProperties> {
+	onInput(event: Event) {
+		this.properties.onInput && this.properties.onInput(event);
+	}
+
+	render() {
```

Comment:
I think we can use `value` + `onInput` instead of `initialValue`. I've been playing with this for the ComboBox. In the parent component that's using either a TextInput or a ComboBox, bind the `value` property on the input to some state in the parent component; then, update the `value` in the `onInput` callback:

**Parent component**
```ts
...
w(TextInput, {
	value: this.inputValue,
	onInput: (value: string) => this.inputValue = value
})
...
```

**Input**
```ts
...
render() {
	const {
		type = 'text',
		value = ''
	} = this.properties;

	return v('input', {
		classes: this.classes(css.textinput).get(),
		type: type,
		value: value,
		oninput: this.onInput
	});
}
...
```